demo/site/src/util/useEscapeKeyPressed.ts

```diff
@@ -0,0 +1,14 @@
+import { useEffect } from "react";
+
+export const useEscapeKeyPressed = (keyPressed: () => void) => {
+    useEffect(() => {
+        const handleKeyDown = (event: KeyboardEvent) => {
+            if (event.key === "Escape") keyPressed();
+        };
+
+        window.addEventListener("keydown", handleKeyDown);
+        return () => {
+            window.removeEventListener("keydown", handleKeyDown);
+        };
+    }, [keyPressed]);
```

Comment:
We could consider simplifying this slightly by using `useEventListener` from `usehooks-ts`: https://usehooks-ts.com/react-hook/use-event-listener